src/platform/intel/cavs/platform.c

```diff
@@ -337,8 +332,18 @@ int platform_init(struct sof *sof)
 
 	shim_write(SHIM_LPSCTL, shim_read(SHIM_LPSCTL));
 
-#elif defined(CONFIG_CANNONLAKE) || defined(CONFIG_ICELAKE) \
-	|| defined(CONFIG_SUECREEK)
+#elif defined(CONFIG_CANNONLAKE)
+
+	/* initialize PM for boot */
+	shim_write(SHIM_CLKCTL,
+		   SHIM_CLKCTL_RHROSCC | /* Request High Performance RING Osc */
+		   SHIM_CLKCTL_OCS_HP_RING | /* Select HP RING Oscillator Clk */
+		   SHIM_CLKCTL_HMCS_DIV2 | /* HP mem clock div by 2 */
+		   SHIM_CLKCTL_LMCS_DIV4 | /* LP mem clock div by 4 */
+		   SHIM_CLKCTL_TCPLCG_EN(0) | /* Core 0 Prevent Local Gating */
+		   SHIM_CLKCTL_TCPLCG_DIS(1));
+
+#elif defined(CONFIG_ICELAKE) || defined(CONFIG_SUECREEK)
 	/* TODO: need to merge as for APL */
 	clock_set_freq(CLK_CPU(cpu_get_id()), CLK_MAX_CPU_HZ);
```

Comment:
Let's make sure that clk freq table and enc are in sync with shim defs. They seem to be scattered, some magics duplicated in clk-map cnd clk headers. (in the next patch).